server/cluster/cluster.go

```diff
@@ -50,24 +50,27 @@ func (s *Server) getConnectionState(cluster appv1.Cluster, errorMessage string)
 			Status:     appv1.ConnectionStatusSuccessful,
 			ModifiedAt: &now,
 		},
+		Version: cluster.ServerVersion,
 	}
 
-	config := cluster.RESTConfig()
```

Comment:
Since connection state is stored in the cluster secret we no longer need to cache it. Instead, we can just return cluster object as is.